### PR TITLE
Remove scp syntax for locations

### PR DESCRIPTION
**Pull request by emilmgeorge**
For #6691.
I have removed the scp part of the regex while keeping the local file part. Also removed the scp syntax from testsuite, docs.
Commits are not yet squashed for easier review.

### CODE REVIEW

**Review by ThomasWaldmann — CHANGES_REQUESTED** [review]
Thanks for the PR!

Looks quite good, but guess you missed some of the docs showing examples in scp style, these need to get fixed also.:
- docs/deployment/central-backup-server.rst
- docs/quickstart.rst
- docs/usage/init.rst

**Comment by ThomasWaldmann** [comment]
src/borg/helpers/parseformat.py:365:13: E126 continuation line over-indented for hanging indent

**Comment by emilmgeorge** [comment]
Thanks, fixed!

**Comment by ThomasWaldmann** [comment]
While you're at it, could you add this:
```
Converting from scp-style repo URLs
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

Borg does not support scp-style repo URLs any more.

Here is how you can convert to URL style:

::

    user@host:path/to/repo             # relative to cwd
    -->
    ssh://user@host:22/./path/to/repo  # relative to cwd
    or (usually the cwd is the home dir after ssh login)
    ssh://user@host:22/~/path/to/repo  # relative to home dir

    user@host:/path/to/repo             # absolute repo path
    -->
    ssh://user@host:22/path/to/repo     # absolute repo path

Notes:

Port 22 is the default, so you can omit the ``:22`` if you like.

If you used some hack to use a non-standard port (which was not directly
supported by the scp-style repo specification), you can now do it that way
with the ``ssh:`` URL and remove the hack (usually in ssh configuration or
given via ``--rsh`` borg option).
```

File: docs/deployment/central-backup-server.rst [hunk 1]
```diff
@@ -81,7 +81,7 @@ The options which are added to the key will perform the following:
 Due to the ``cd`` command we use, the server automatically changes the current
 working directory. Then client doesn't need to have knowledge of the absolute
 or relative remote repository path and can directly access the repositories at
-``<user>@<host>:<repo>``.
+``ssh://<user>@<host>/<repo>``.
```
Comment (ThomasWaldmann) [inline docs/deployment/central-backup-server.rst:84]: Hmm, considering the sentence in line 82/83, this needs to stay a relative-to-cwd path, just use the ssh url syntax.

File: docs/usage/general/repository-urls.rst.inc [hunk 2]
```diff
@@ -52,3 +36,31 @@ to use the default - it will be read from BORG_REPO then.
...
+
+    user@host:path/to/repo             # relative to cwd
+    -->
+    ssh://user@host:22/./path/to/repo  # relative to cwd
+    or (usually the cwd is the home dir after ssh login)
+    ssh://user@host:22/~/path/to/repo  # relative to home dir
+
+    user@host:/path/to/repo             # absolute repo path
+    -->
+    ssh://user@host:22/path/to/repo     # absolute repo path
```
Comment (ThomasWaldmann) [inline docs/usage/general/repository-urls.rst.inc:57]: can you check/fix column of the `#` being the same for these lines?

File: docs/usage/init.rst [hunk 3]
```diff
@@ -15,8 +15,8 @@ Examples
 
     # Remote repository (accesses a remote borg via ssh)
     # repokey: stores the (encrypted) key into <REPO_DIR>/config
-    $ borg init --encryption=repokey-aes-ocb user@hostname:backup
+    $ borg init --encryption=repokey-aes-ocb ssh://user@hostname/backup
 
     # Remote repository (accesses a remote borg via ssh)
     # keyfile: stores the (encrypted) key into ~/.config/borg/keys/
-    $ borg init --encryption=keyfile-aes-ocb user@hostname:backup
+    $ borg init --encryption=keyfile-aes-ocb ssh://user@hostname/backup
```
Comment (ThomasWaldmann) [inline docs/usage/init.rst:22]: relative-to-cwd needed

**Review by ThomasWaldmann — CHANGES_REQUESTED** [review]
discovered some more issues. but guess we're done after fixing these.

File: docs/deployment/central-backup-server.rst (hunk 1 above)
Comment (emilmgeorge) [reply docs/deployment/central-backup-server.rst]: Done

**Review by ThomasWaldmann — APPROVED** [review]
LGTM

**Comment by ThomasWaldmann** [comment]
Thanks!

**Comment by ThomasWaldmann** [comment]
Crap, I was a little fast with merging this.

The idea was to merge this IF the other breaking changes are also merged.
But anyway, either we do that or we'll revert this change later again.

**Pull request merged**